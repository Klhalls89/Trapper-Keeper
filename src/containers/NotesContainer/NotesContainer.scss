
.notes-container {
  display: grid;
  grid-auto-flow: dense;
  grid-auto-columns: repeat(5 1fr);
}