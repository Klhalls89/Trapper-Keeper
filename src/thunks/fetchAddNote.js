import { addNote, setLoading, setError } from '../actions/index';

export const fetchAddNote = (note) => {
  return async (dispatch) => {
    const url = 'http://localhost:3001/api/v1/notes';
    const body = {
      title: note.title,
      list: note.list,
      id: note.id
    }
    const options = {
      method: 'POST',
      headers: { 'Content-type': 'application/json' },
      body: JSON.stringify(body)
    }

    try {
      dispatch(setLoading(true));

      const response = await fetch(url, options);

      if (!response.ok) {
        throw Error(response.statusText);
      }

      const note = await response.json();
      
      dispatch(setLoading(false));
      dispatch(addNote(note));
    } catch (error) {
      dispatch(setError(error.message));
    }
  }
}