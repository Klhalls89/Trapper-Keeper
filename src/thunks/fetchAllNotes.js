import { setLoading, addAllNotes, setError } from '../actions';

export const fetchAllNotes = () => {
  return async (dispatch) => {
    const url = 'http://localhost:3001/api/v1/notes';

    try {
      dispatch(setLoading(true));

      const response = await fetch(url);

      if(!response.ok) {
        throw Error(response.statusText);
      }

      const notes = await response.json();

      dispatch(addAllNotes(notes));
      dispatch(setLoading(false));
    } catch (error) {
      dispatch(setError(error.message));
    }
  }
}