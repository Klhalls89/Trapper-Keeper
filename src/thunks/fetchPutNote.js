import { setError, updateNote, setLoading } from '../actions';

export const fetchPutNote = (note) => {
  return async (dispatch) => {
    const id = note.id;
    const body = {
      title: note.title,
      list: note.list,
      id: note.id
    }
    const options = {
      method: 'PUT',
      headers: { 'Content-type': 'application/json' },
      body: JSON.stringify(body)
    }
    const url = `http://localhost:3001/api/v1/notes/${id}`;
  
    try {
      dispatch(setLoading(true));
      
      const response = await fetch(url, options);
      
      if (!response.ok) {
        throw Error(response.statusText);
      }
      
      dispatch(setLoading(false));
      dispatch(updateNote(body));
    } catch(error) {
      dispatch(setError(error.message));
    }
  }
}