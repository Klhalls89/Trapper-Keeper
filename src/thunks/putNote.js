import { setError, updateNote, setLoading } from '../actions'

export const putNote = (note) => {
  return async (dispatch) => {
    const id = note.id
    const body = {
      title: note.title,
      list: note.list,
      id: note.id
    }
    const options = {
      method: 'PUT',
      headers: { 'Content-type': 'application/json' },
      body: JSON.stringify(body)
    }
    const url = `http://localhost:3001/api/v1/notes/${id}`
  
    try {
    dispatch(setLoading(true))
    dispatch(updateNote(body));

    const response = await fetch(url, options);

    if (!response.ok) {
      throw Error(response.statusText)
    }
    
    dispatch(setLoading(false))

    const status = await response.json();
    console.log('status',status)


    } catch(error){
      dispatch(setError(error.message))
    }
  }
}