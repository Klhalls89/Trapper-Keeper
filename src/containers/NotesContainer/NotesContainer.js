import React, { Component } from 'react';
import { connect } from 'react-redux';
import { fetchAllNotes } from '../../thunks/fetchAllNotes';
import { fetchDeleteNote } from '../../thunks/fetchDeleteNote';
import Note from '../../components/Note/Note';
import NoteForm from '../NoteForm/NoteForm';
import {Link} from 'react-router-dom';

export class NotesContainer extends Component {
  componentDidMount() {
    this.props.fetchAllNotes();
  }

  render() {
    let displayNotes;
    let { notes, location} = this.props;
    let notePopup;

    if (notes.length) {  
      displayNotes = notes.map(note => 
        <Note {...note} key={note.id} fetchDeleteNote={this.props.fetchDeleteNote}/>
      );
    } else {
      displayNotes = <div className='empty-notes'>
        <i class="fas fa-edit"></i>
        <p>Notes will display here</p>
      </div>
    }

    if (location.pathname === "/new-note" ) {
      notePopup = 
        <div className="popup-background">
          <NoteForm />
        </div>
    } 

    if (location.pathname.includes('/notes/')) {
      const id = location.pathname.split("/")[2]
      const match = notes.find(note => {
        return note.id === parseInt(id);
      });

      if (match !== undefined) {
        notePopup = 
        <div className="popup-background">
          <NoteForm note={match} />
        </div>
      }
    }

    return (
      <div className="main-container">
        <Link to="/new-note">
          <i className="fas fa-plus add-btn"></i>      
        </Link>
        <section className="notes-container">
          {displayNotes}
        </section>
        {notePopup}
      </div>
    );
  }
}

export const mapStateToProps = state => ({
  notes: state.notes
});

export const mapDispatchToProps = dispatch => ({
  fetchAllNotes: () => dispatch(fetchAllNotes()),
  fetchDeleteNote: id => dispatch(fetchDeleteNote(id))
});

export default connect(mapStateToProps, mapDispatchToProps)(NotesContainer);
