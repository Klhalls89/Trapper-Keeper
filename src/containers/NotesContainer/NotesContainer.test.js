import React from 'react'
import NotesContainer from './NotesContainer'
import { fetchAllNotes } from '../../thunks/fetchAllNotes';
import { shallow } from 'enzyme'

jest.mock('../../thunks/fetchAllNotes')

describe('NotesContainer', () => {

  it('should match snapshot', () => {
   expect(NotesContainer).toMatchSnapshot();
  })

  describe('componentDidMount', () => {

    fetchAllNotes.mockImplementation(() => {})

    it('should call fetchAllNotes', () => {
     expect(fetchAllNotes).toHaveBeenCalled()
    })

  })

  describe('Render', () => {

    it('should return displayNotes', () => {

    })

    it('should return notePopup', () => {

    })

  })

  describe('mapStateToProps', () => {

    it('should map state to props', () => {

    })

  })

  describe('mapDispatchToProps', () => {

    it('should map dispatch to props', () => {

    })
    
  })

})