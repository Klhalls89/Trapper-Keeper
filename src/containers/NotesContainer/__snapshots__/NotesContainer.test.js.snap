// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NotesContainer should match snapshot 1`] = `
Object {
  "$$typeof": Symbol(react.memo),
  "WrappedComponent": [Function],
  "compare": null,
  "displayName": "Connect(NotesContainer)",
  "type": [Function],
}
`;
